js/apps/admin-ui/src/user/UserForm.tsx

```diff
@@ -133,284 +137,217 @@ export const UserForm = ({
       fineGrainedAccess={user?.access?.manage}
       className="pf-u-mt-lg"
     >
-      {open && (
-        <GroupPickerDialog
-          type="selectMany"
-          text={{
-            title: "selectGroups",
-            ok: "join",
-          }}
-          canBrowse={isManager}
-          onConfirm={(groups) => {
-            user?.id ? addGroups(groups || []) : addChips(groups || []);
-            setOpen(false);
-          }}
-          onClose={() => setOpen(false)}
-          filterGroups={selectedGroups}
+      <FormProvider {...form}>
+        {open && (
+          <GroupPickerDialog
+            type="selectMany"
+            text={{
+              title: "selectGroups",
+              ok: "join",
+            }}
+            canBrowse={isManager}
+            onConfirm={(groups) => {
+              user?.id ? addGroups(groups || []) : addChips(groups || []);
+              setOpen(false);
+            }}
+            onClose={() => setOpen(false)}
+            filterGroups={selectedGroups}
+          />
+        )}
+        {user?.id && (
+          <>
+            <FormGroup label={t("id")} fieldId="kc-id" isRequired>
+              <KeycloakTextInput
+                id={user.id}
+                aria-label={t("userID")}
+                value={user.id}
+                readOnlyVariant="default"
+              />
+            </FormGroup>
+            <FormGroup
+              label={t("createdAt")}
+              fieldId="kc-created-at"
+              isRequired
+            >
+              <KeycloakTextInput
+                value={formatDate(new Date(user.createdTimestamp!))}
+                id="kc-created-at"
+                aria-label={t("createdAt")}
```

Comment:
There should be no need to add this `aria-labrl` as the field is already associated with a label through the form group.